### DISCUSSION TITLE
Should we remove the `setup` and `cleanup` commands in windows and macOS

### DISCUSSION THREAD

**Discussion by anjannath** [crc-org/crc#2377] [Q&A]
This has been brought up earlier by team members, i wanted to have it here to have kind of a plan, and also find out what will be the end user experience for non-installer users on these platforms.

With the macOS pkg and msi for windows we can run the `crc setup` as a post install step or move some of the setup work to the installer provided ways of doing them. (e.g creating shortcut for tray in start up folder, putting `oc` on path can be done from the msi wix file)

`setup` and `cleanup` will no longer be needed then on Windows and macOS.

**Comment by cfergeau** [comment]
This could be a relatively longer term discussion to be had, I think it depends on various steps:
- no more embedded binaries on windows/mac
- only usermode networking on windows/mac
- having the installers perform setup/cleanup

Once we have all of this, we should have a clearer idea whether this is a realistic option, or if we have to keep them for some reasons we don't know of yet. In an ideal world, yes we probably could do that, but for now, I don't think we can easily make a decision regarding this.
